bin/hdfcoinc/pycbc_page_snrratehist

```diff
@@ -26,17 +28,41 @@ parser.add_argument('--output-file')
 parser.add_argument('--bin-size', type=float, default=0.1)
 parser.add_argument('--x-min', type=float, default=8.0)
 parser.add_argument('--trials-factor', type=int, default=1)
-parser.add_argument('--cumulative', action='store_true')
+parser.add_argument('--h-iter-inc-background', type=int, default=None,
+                    help='Indicate which inclusive background to plot '
+                         'with the foreground triggers if there were '
+                         'any hierarchical removals done. Choosing 0 '
+                         'indicates plotting the inclusive background '
+                         'prior to any hierarchical removals. Choosing 1 '
+                         'indicates the inclusive background after the '
+                         'loudest foreground trigger was removed. '
+                         '[default=None]')
 parser.add_argument('--closed-box', action='store_true',
-      help="Make a closed box version that excludes foreground triggers")
+                    help='Make a closed box version that excludes '
+                         'foreground triggers')
 args = parser.parse_args()
 
+logging.info('Read in the data')
+f = h5py.File(args.trigger_file, 'r')
+
+# Determine which inclusive background to plot.
+if args.h_iter_inc_background != None:
+    h_inc_back_num = args.h_iter_inc_background
+    if f.attrs['hierarchical_removal_iterations'] is not None:
+        h_iterations = f.attrs['hierarchical_removal_iterations']
+        if h_inc_back_num > h_iterations:
+            raise ValueError('User requested inclusive background after '
+                             'hierarchical removals that does not exist!')
+else :
+    if f.attrs['hierarchical_removal_iterations'] is not None:
+        h_inc_back_num = f.attrs['hierarchical_removal_iterations']
+        h_iterations = h_inc_back_num
+    else :
+        h_inc_back_num = 0
+
```

Comment:
Ugh, unfortunately that line occurs some 6 times in the code. `f.attrs['hierarchical_removal_iterations']` is always populated when `pycbc_coinc_statmap` runs (cf. https://github.com/ligo-cbc/pycbc/blob/master/bin/hdfcoinc/pycbc_coinc_statmap#L336). However, if one wants to use old hdf coinc trigger files, then you're right this code will fall over. I'll implement it the way that you suggested to avoid that failure.